presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java

```diff
@@ -479,7 +479,18 @@ public Expression visitFunctionCall(FunctionCall node, Context<C> context)
                 }
             }
 
+            Optional<Expression> filter = node.getFilter();
+            if (filter.isPresent()) {
+                Expression filterExpression = filter.get();
+                Expression newFilterExpression = rewrite(filterExpression, context.get());
+                filter = Optional.of(newFilterExpression);
+            }
+
             Optional<Window> rewrittenWindow = node.getWindow();
+            //filter with window function is not supported yet
+            if (filter.isPresent() && node.getWindow().isPresent()) {
+                throw new UnsupportedOperationException("not yet implemented: filters with window functions");
```

Comment:
We need to return a proper `NOT_SUPPORTED` error to the user with a good message like "FILTER is not yet supported for window functions".

The parser is not the right place to do this. It should happen somewhere in the analyzer.